doc/branches-explained.md

```diff
@@ -0,0 +1,39 @@
+# HDF5 Git Branching Model Explained
+
+This document desribes current HDF5 branches. 
+
+Branches are tested nightly and testing results are available at https://cdash-internal.hdfgroup.org/ and https://cdash.hdfgroup.org/. Commits that break daily testing should be fixed by 3:00 pm Central time or reverted.  We encourgae code contributors to check the status of their commits. If you have any questions, please conrtact lrknox@hdfgroup.org.
+ 
+## `develop`
+Develop is the main branch whose source code always reflects a state with the latest delivered development changes for the next major release of HDF5. 
+This is also considered the integration branch, as **all** new features are integrated into this branch from respective feature branches. 
+
+## `Maintenance branches`
+
+Each currently supported release-line of HDF5 (e.g. 1.8.x, 1.10.x, 1.12.x) has a support branch with the name 1_8, 1_10, 1_12. 
+Maintenance branches are similar to the develop branch, except the source code in a maintenance branch always reflects a state 
+with the latest delivered development changes for the next **maintenance** release of that particular supported release-line of HDF5. 
+**Some** new features will be integrated into a release maintenance branch, depending on whether or not those features can be 
+introduced in minor releases.  Maintenance branches are removed when a release-line is retired from support.
+
+## `feature/*`
+Feature branches are temporary branches used to develop new features in HDF5.
+Feature branches branch off of develop and exist as long as the feature is under development. 
+When the feature, the branch is merged back into develop, as well as any support branches in which the change will be included, and then the feature branch is removed.
+
+## `release/*`
+Release branches are used to prepare a new production release. They are primarily used to allow for last minute dotting of is and crossing of ts 
+(things like setting the release version, finalizing release notes, et cetera) and do not include new development. 
+They are created from maintenance branch at the time of the maintenance release and have 
+names 1_8_N, 1_10_N, 1_12_N, where N is minor release number. Once the release is  done it is tagged. 
```

Comment:
Ditto. '... is the minor release number.'

Reply:
Thank you fro catching typos. I am not sure how I managed to miss so many. I will check,  I think N is called "release number", 8, 10, 12 - minor release, and 1 major. We do have a problem with our versioning and naming conventions, because we call 8,10,12 - major releases...

Reply:
Done